src/writer.rs

```diff
@@ -782,3 +808,24 @@ mod indentation {
         );
     }
 }
+
+#[test]
+fn element_writer_newline() {
+    let mut buffer = Vec::new();
+    let mut writer = Writer::new_with_indent(&mut buffer, b' ', 4);
+
+    writer
+    .create_element("element")
+    .with_attribute(("first", "1"))
+    .with_attribute(("second", "2"))
+    .new_line()
+    .with_attribute(("third", "3"))
+    .with_attribute(("fourth", "4"))
+    .write_empty().expect("write tag failed");
+
+    assert_eq!(
+        std::str::from_utf8(&buffer).unwrap(),
+        r#"<element first="1" second="2"
+ third="3" fourth="4"/>"#
```

Comment:
The correct output of this test is:
```suggestion
        r#"<element first="1" second="2"
         third="3" fourth="4"/>"#
```
I think, this is because of line 387 above.